jwst/lib/set_telescope_pointing.py

```diff
@@ -1516,8 +1517,12 @@ def populate_model_from_siaf(model, siaf):
 
     # For TSO exposures, also populate XREF_SCI/YREF_SCI keywords,
     # which are used by the Cal pipeline to determine the
-    # location of the source
-    if model.meta.visit.tsovisit:
+    # location of the source.
+    # Note that we use a combination of the is_tso function and
+    # a check on EXP_TYPE, because there are rare corner cases
+    # where EXP_TIME=NRC_TSGRISM, TSOVISIT=False, NINTS=1, which
+    # normally return False, but we want to treat it as TSO anyway.
+    if is_tso(model) or model.meta.exposure.type.lower() in ['nrc_tsimage', 'nrc_tsgrism']:
```

Comment:
Big clarification question: `is_tso` enforces NINTS >1. Conceptually, does TSO data have to have NINTS>1? I.e., if this particular check was ignored, and all other indicators say `is_tso` is True, that any code would be correct in continuing processing such data as TSO?

Reply:
Unfortunately we've now gotten into gray territory where the answer to NINTS > 1 is "it depends". There are some cases where it's OK and even desired to allow NINTS=1 and others where it isn't. Hence this code essentially overrides `is_tso` in the particular case of NRC_TSGRISM, TSOVISIT=F, and NINTS=1.

Reply:
Otherwise, you're right that it would've been far simpler to just change `is_tso` itself to allow NINTS=1.